### PR TITLE
Minor: Add tests for using FilterExec when parquet was pushed down

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Part of https://github.com/apache/datafusion/issues/4028

## Rationale for this change

While reviewing https://github.com/apache/datafusion/pull/12135 from @itsjunetime I wanted to make sure everything was hooked up end to end and that the appropriate plan changes were done. 

I figured I would write such tests as part of my review, though for some reason the PR doesn't actually improve these plans (I will comment on that separately). 

## What changes are included in this PR?
This adds "end to end" tests that show what should happen when parquet filter pushdown is enabled (specifically the FilterExec should be gone)

## Are these changes tested?
yes, only tests

## Are there any user-facing changes?
No, only tests

### CODE REVIEW

File: datafusion/sqllogictest/test_files/parquet.slt [hunk 1]
```diff
@@ -202,6 +202,65 @@ SELECT id, CAST(string_col AS varchar) FROM alltypes_plain
...
+
+query TT
+EXPLAIN SELECT id FROM alltypes_plain WHERE id > 3;
+----
+logical_plan
+01)Filter: alltypes_plain.id > Int32(3)
+02)--TableScan: alltypes_plain projection=[id], partial_filters=[alltypes_plain.id > Int32(3)]
+physical_plan
+01)CoalesceBatchesExec: target_batch_size=8192
+02)--FilterExec: id@0 > 3
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/parquet.slt:255]: The goal of https://github.com/apache/datafusion/issues/4028 is to remove this FilterExec

File: datafusion/sqllogictest/test_files/parquet_filter_pushdown.slt [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+## Create table without pushdown
+statement ok
+CREATE EXTERNAL TABLE t(a varchar, b int, c float) STORED AS PARQUET
+LOCATION 'test_files/scratch/parquet_filter_pushdown/parquet_table/';
+
+## Create table with pushdown enabled (note the pushdown setting is part of the table,
+## but is copied from the session settings, so we need to set it here)
+
+statement ok
+set datafusion.execution.parquet.pushdown_filters = true;
```
Comment (Jefffrey) [inline datafusion/sqllogictest/test_files/parquet_filter_pushdown.slt:50]: Is it worth setting this to false before creation of `t` explicitly, or reckon it's unnecessary and fine to rely on defaults?

**Review by Jefffrey — APPROVED** [review]
Test case makes sense to me :+1:

File: datafusion/sqllogictest/test_files/parquet_filter_pushdown.slt (hunk 2 above)
Comment (alamb) [reply datafusion/sqllogictest/test_files/parquet_filter_pushdown.slt:55]: It is a good idea to make it explicit -- I will do so and add a comment about that

**Comment by alamb** [comment]
Thanks again @Jefffrey  for your review

**Pull request merged**